core/src/main/java/org/jclouds/concurrent/config/ExecutorServiceModule.java

```diff
@@ -54,6 +58,42 @@
 @ConfiguresExecutorService
 public class ExecutorServiceModule extends AbstractModule {
 
+   private static final Method CREATE_STL;
+   private static final Constructor<SimpleTimeLimiter> CONSTRUCT_STL;
+   static {
+      Method create = null;
+      Constructor ctor = null;
+      try {
+         create = SimpleTimeLimiter.class.getDeclaredMethod("create", ExecutorService.class);
+      } catch (NoSuchMethodException nsme) {
+         try {
+            ctor = SimpleTimeLimiter.class.getConstructor(ExecutorService.class);
+         } catch (NoSuchMethodException nsme2) {
+            throw Throwables.propagate(nsme2);
+         }
+      }
+      CREATE_STL = create;
+      CONSTRUCT_STL = ctor;
+   }
+
+   static SimpleTimeLimiter createSimpleTimeLimiter(ExecutorService executorService) {
+      try {
+         if (CREATE_STL != null) {
+            return (SimpleTimeLimiter) CREATE_STL.invoke(null, executorService);
+         } else if (CONSTRUCT_STL != null) {
+            return CONSTRUCT_STL.newInstance(executorService);
+         }
+         // fall through
+      } catch (IllegalAccessException iae) {
+         // fall through
+      } catch (InstantiationException ie) {
+         // fall through
+      } catch (InvocationTargetException ite) {
+         // fall through
```

Comment:
It would be better to propagate the cause of these exceptions.

Reply:
Done